valet

```diff
@@ -81,16 +75,36 @@ then
     ARCH=$(uname -m)
 
     if [[ $ARCH == 'arm64' ]]; then
-        sudo -u "$USER" "$DIR/bin/ngrok-arm" http "$HOST.$TLD:$PORT" -host-header=rewrite $PARAMS
+        "$DIR/bin/ngrok-arm" http "$HOST.$TLD:$PORT" -host-header=rewrite $PARAMS
     else
-        sudo -u "$USER" "$DIR/bin/ngrok" http "$HOST.$TLD:$PORT" -host-header=rewrite $PARAMS
+        "$DIR/bin/ngrok" http "$HOST.$TLD:$PORT" -host-header=rewrite $PARAMS
     fi
 
     exit
 
+# Proxy PHP commands to the "php" executable on the isolated site
+elif [[ "$1" = "php" ]]
+then
+    $(php "$DIR/cli/valet.php" which-php) "${@:2}"
+
+    exit
+
+# Proxy Composer commands with the "php" executable on the isolated site
+elif [[ "$1" = "composer" ]]
+then
+    $(php "$DIR/cli/valet.php" which-php) $(which composer) "${@:2}"
```

Comment:
Are we going to run into any trouble where we're running a lower version of PHP than Composer is globally updated to expect? If so, what's the error we see?

Reply:
In that case, **any valet command** would fail with a similar error.  

```bash
valet -V

Fatal error: Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 8.1.0". You are running 7.0.33. in /Users/user/.composer/vendor/composer/platform_check.php on line 24
```

Reply:
btw, if you are asking about the other way around. Like if isolated `php@7.0` would be able to run composer commands when the global composer is updated to `php@8.1`. In that case, I can see it works fine.